benchmark/worker/benchmark_client.go

```diff
@@ -54,11 +54,12 @@ var (
 )
 
 type benchmarkClient struct {
-	closeConns    func()
-	stop          chan bool
-	mu            sync.RWMutex
-	lastResetTime time.Time
-	histogram     *stats.Histogram
+	closeConns       func()
+	stop             chan bool
+	lastResetTime    time.Time
+	histogramOptions stats.HistogramOptions
+	mutexes          []*sync.Mutex
+	histograms       []*stats.Histogram
```

Comment:
can you do 

hs   []*lockingHistogram

where

type lockHistogram {
  mu  sync.Mutex
  h     *stats.Histogram
}